includes/admin/upgrades/upgrade-functions.php

```diff
@@ -1349,4 +1364,232 @@ function edd_remove_legacy_discounts() {
+					'token'        => isset( $post_meta['_edd_log_token'] ) ? $post_meta['_edd_log_token'] : 'public',
+					'version'      => $post_meta['_edd_log_version'],
+					'time'         => $post_meta['_edd_log_time'],
+					'request'      => $old_log->post_excerpt,
+					'error'        => $old_log->post_content,
+					'date_created' => $old_log->post_date,
+				);
+
+				$new_log_id = EDD()->api_request_logs->insert( $log_data );
+
+				if ( ! empty( $new_log_id ) ) {
+					add_post_meta( $old_log->ID, '_edd_log_migrated_id', $new_log_id );
+				}
+			} else {
+				$post = new WP_Post( $old_log->ID );
+
+				$log_data = array(
+					'object_id'   => $post->post_parent,
+					'object_type' => 'download',
+					'type'        => $old_log->slug,
+					'title'       => $old_log->post_title,
+					'message'     => $old_log->post_content
+				);
+
+				$meta            = get_post_custom( $old_log->ID );
+				$meta_to_migrate = array();
+
+				foreach ( $meta as $key => $value ) {
+					$meta_to_migrate[ $key ] = maybe_unserialize( $value[0] );
+				}
+
+				$new_log_id = EDD()->logs->insert( $log_data );
+				$new_log = new EDD\Logs\Log( $new_log_id );
+				$new_log->add_meta( 'legacy_id', $old_log->ID );
+
+				if ( ! empty( $meta_to_migrate ) ) {
+					foreach ( $meta_to_migrate as $key => $value ) {
+						$new_log->add_meta( $key, $value );
+					}
+				}
+			}
+
+			edd_debug_log( $old_log->ID . ' successfully migrated to ' . $new_log_id );
+		}
+
+		edd_debug_log( 'Step ' . $step . ' of logs migration complete' );
+
+		$step++;
+		$redirect = add_query_arg( array(
+			'page'        => 'edd-upgrades',
+			'edd-upgrade' => 'logs_migration',
+			'step'        => $step,
+			'number'      => $number,
+			'total'       => $total
+		), admin_url( 'index.php' ) );
+
+		wp_safe_redirect( $redirect );
+		exit;
+	} else {
+		update_option( 'edd_version', preg_replace( '/[^0-9.].*/', '', EDD_VERSION ) );
+		edd_set_upgrade_complete( 'migrate_logs' );
+		delete_option( 'edd_doing_upgrade' );
+
+		edd_debug_log( 'All old logs migrated, upgrade complete.' );
+
+		wp_redirect( admin_url() );
+		exit;
+	}
+}
+add_action( 'edd_logs_migration', 'edd_logs_migration' );
+
+/**
+ * Removes legacy logs data.
+ *
+ * @since 3.0
+ */
+function edd_remove_legacy_logs() {
+	global $wpdb;
+
+	if ( ! current_user_can( 'manage_shop_settings' ) ) {
+		return;
+	}
+
+	ignore_user_abort( true );
+	set_time_limit( 0 );
+
+	$log_ids = $wpdb->get_results( "SELECT ID FROM $wpdb->posts WHERE post_type = 'edd_log'" );
+	$log_ids = wp_list_pluck( $log_ids, 'ID' );
+	$log_ids = implode( ', ', $log_ids );
+
+	edd_debug_log( 'Beginning removal of legacy logs' );
+
+	if ( ! empty( $log_ids ) ) {
+		$delete_posts_query = "DELETE FROM $wpdb->posts WHERE ID IN ({$log_ids})";
+		$wpdb->query( $delete_posts_query );
+		$delete_postmeta_query = "DELETE FROM $wpdb->postmeta WHERE post_id IN ({$log_ids})";
```

Comment:
Do we know for certain this will work on sites that have 2,000,000+ log entries?

Reply:
I think this query will be better if I change it to `DELETE FROM $wpdb->posts WHERE post_type = 'edd_log'`

Reply:
I agree